### ISSUE TITLE
Sybil Attacker Report

### ISSUE THREAD

**Issue by GreyPK** [hop-protocol/hop-airdrop#191]
# Related Addresses

59 related addresses. 
```
0x50a50e86767809b2e90327cf81f98a2668d0ce46
0xe4658066c46dbdd6689594cd35970fc44d44e8c5
0xe5c38308dfdd1e582f648dd000fca79d29c75981
0xd2d90b379bd84962f1343a9f86b919701dace09c
0x5a679d44eb779165fa112139bc1498351bd644b0
0xba1d58004f3d5e268c40619764872f4f9892692f
0x6c9618cd9400737f183ee5d89cfcaad7423220d8
0x8cd983b2cc065d64b4d3e13aca9447a5aca15d4e
0x0b8181cc4f17d4243d9c88779bb713265791e903
0xea9ebe9aeff712e20edcbd47b238655631bd226e
0x7955e9278b6c4d3bb0948f4e776936c17a794354
0xad1cb13cb3db4694c41a1a91bf4e2e37ef4c2ad1
0x4939ca5dd5364bad45321b90aece36855d823ce5
0xe5865632439987c02d6634ed8335948859fead59
0x58371039b7314b7dd9aae1c43c9fc2b001deb996
0x20b8152ff80c8e7e8a56d47f428a08c054a8ee56
0xab2227c706ccf94bfcfa9b55f0933b3803c0cda9
0x6509f59753b1239322760a6a995b5465a7ca501a
0x89f360b9deae4e0f3ec2bfda31e212e2d0567776
0xa018924e0d156a13f888494659450e6fcaa9ac28
0xdf727d07ebd49237d3f1f34f79b63860139a6c7c
0xbb983df78c0a7fde3c58dc0caca3a2cd5c7868e8
0xe01ba9d1143306e29fbd39fbd5e1cf8cb44a864d
0xdb77dbc91b48f758a332b8961407ca1d22c8292c
0x1bd89d24d3435ad9290e528e4f56472d3468de00
0x7fefd2303b2260abcd97609d48cdd5b61fbb5fdc
0xf8389b647ec7331c64b3c3c91fce93cffb7b075b
0xe15967327932a7ae8d6334f079bd889e20c608b2
0xb5c9b2f2043230b4b86b2884724263b93a299009
0xbb0d240bc794514fc84287ca4ad77282810734cb
0x54cb7407b7a6eca9715bfb3a467ff056acef2d01
0xecf08e10706832af66adec7702aa9eff0fdc878d
0xe2b09bf03404e09007325ebf49b8633848121b5d
0x0068c58783a29256260bb807dfad4807b7193087
0x0331de7345722f62fb31dee19cf6891e13544c63
0xe9c3a8a0ee3e5faf36bbf02d228cf6a2a767213e
0xa57ea970d1f25975640f0d81bf613287b6aa62ea
0xe2118286257a56b864039bbfb90bf3658d13c3d1
0x62d644a6e42e7f93c9704cf456f1fc3d15757ce4
0xea00e2ab1c6878e04ba7f0e310eeb78836bee076
0xe55082fd99bbbd79ec991309417c756e9c1fede7
0xed10ce0f55287e96fa8ef6809b2c948981117f52
0x3fa1b088af50d6b04cf043d46d2bc1e3f239700b
0x7562207875419a6951da5663681ecf611d94f576
0x0495a14956a926f1d8f8176822455bf3476178cd
0xa578487a88f3ef1c3d9788cf0ceb3fb5460f2ba2
0x994831005283888124443a687cf3731ab59428e7
0xc71cdefcdef9572417a4a923e4ca80d3ff237327
0x97c961dfd3fffc4e2980f7ec6402e49640683d4f
0x8c849d6cedf61bd72a0672a54f1a99d1c339e5ab
0x7f97ba1c342f713420a4625772b88d4d38956bd2
0x14ccf97bcd25e5c875b367e93cbfde93894db98f
0x7c7279711d010e84089df83dc8a3ec91656f4acd
0xe5af587795c94162028352a816db691ebc7ad71f
0x5684f23a3a8f9a367612145ac05d0dad96ccf5da
0x2316a6b76dbf0ba68073d972d5a42a050501c585
0x8e5ed166aa89a3c98719aeb7348ddcceef7e39bb
0x40b64b354f2079c8aac9320f8f2b386fffca423c
0xbfa31ebe9b1112c64278112090ba2e2130418704
```

# Reasoning

1) An elligible for the airdrop Address 0x50a50e86767809b2e90327cf81f98a2668d0ce46 send MATIC to these 58 addresses that elligible for the airdrop. These transactions can find here: https://polygonscan.com/txs?a=0x50a50e86767809b2e90327cf81f98a2668d0ce46&f=2&ps=100&p=8
2) All 58 wallets first used HOP with SwapAndSend in almost same day (20-23 Sep 2021) with 900-1300 USDT (sometimes about 200 USDT).
3) After all 58 wallets used HOP SwapAndSend in 23-27 Nov 2021 with 4 transactions with volume about $100.

Examples for first 10 wallets:
0xe4658066c46dbdd6689594cd35970fc44d44e8c5 first used HOP swapAndSend 1043 USDT Sep-20-2021 https://polygonscan.com/tx/0x21d356067e18c3f0e9494263890d51e06e06e9d5a3456296ce70fdaa18117f0e
After used HOP SwapAndSend USDT 4 times (20, 20, 23, 29 USDT) from Nov-24-2021 to Nov-27-2021
https://polygonscan.com/txs?a=0xe4658066c46dbdd6689594cd35970fc44d44e8c5&f=2&ps=100&p=1

0xe5c38308dfdd1e582f648dd000fca79d29c75981 first used HOP swapAndSend 1068 USDT Sep-20-2021 https://polygonscan.com/tx/0x9524a377fe715c1905c35f8d31090ab30ddc1645c95a8361c3ac4809cd18c39b
After used HOP SwapAndSend USDT 4 times (22, 23, 29, 24 USDT) from Nov-23-2021 to Nov-27-2021
https://polygonscan.com/txs?a=0xe5c38308dfdd1e582f648dd000fca79d29c75981&f=2&ps=100&p=1

0xd2d90b379bd84962f1343a9f86b919701dace09c first used HOP swapAndSend 1094 USDT Sep-20-2021 https://polygonscan.com/tx/0x4c71a311a2c2f5d83547cd9f4d018644bf0806f265352d1cfc5e33f496078d9d
After used HOP SwapAndSend USDT 4 times (28, 30, 27, 23 USDT) from Nov-23-2021 to Nov-27-2021
https://polygonscan.com/txs?a=0xd2d90b379bd84962f1343a9f86b919701dace09c&f=2&ps=100&p=1

0x5a679d44eb779165fa112139bc1498351bd644b0 first used HOP swapAndSend 1073 USDT Sep-20-2021 https://polygonscan.com/tx/0xb7e36ea6a75cbc890e02f9990e0c1eae7b58efa96f5b7ddadd1ae89119e5c5f6
After used HOP SwapAndSend USDT 4 times (22, 20, 27, 29 USDT) from Nov-23-2021 to Nov-27-2021
https://polygonscan.com/txs?a=0x5a679d44eb779165fa112139bc1498351bd644b0&f=2&ps=100&p=1

0xba1d58004f3d5e268c40619764872f4f9892692f first used HOP swapAndSend 1056 USDT Sep-20-2021 https://polygonscan.com/tx/0xd808e0230dc223e1fd328c99c57fa0994fc55205a5bb0d31d94e03fd026a99d8
After used HOP SwapAndSend USDT 4 times (25, 26, 28, 26 USDT) from Nov-23-2021 to Nov-27-2021
https://polygonscan.com/txs?a=0xba1d58004f3d5e268c40619764872f4f9892692f&f=2&ps=100&p=1

0x6c9618cd9400737f183ee5d89cfcaad7423220d8 first used HOP swapAndSend 1034 USDT Sep-20-2021 https://polygonscan.com/tx/0x6bf272a915621684f71b3b934c30442fceb0d4ca9b58dca12cf35dd2a2d8aa4b
After used HOP SwapAndSend USDT 4 times (24, 20, 24, 20 USDT) from Nov-23-2021 to Nov-27-2021
https://polygonscan.com/txs?a=0x6c9618cd9400737f183ee5d89cfcaad7423220d8&f=2&ps=100&p=1

0x8cd983b2cc065d64b4d3e13aca9447a5aca15d4e first used HOP swapAndSend 1101 USDT Sep-20-2021 https://polygonscan.com/tx/0x0b673406a41cebf593268bb286633297cf8790975379cfeb5a28118f98ec531f
After used HOP SwapAndSend USDT 4 times (22, 24, 24, 22 USDT) from Nov-23-2021 to Nov-27-2021
https://polygonscan.com/txs?a=0x8cd983b2cc065d64b4d3e13aca9447a5aca15d4e&f=2&ps=100&p=1

0x0b8181cc4f17d4243d9c88779bb713265791e903 first used HOP swapAndSend 1077 USDT Sep-20-2021 https://polygonscan.com/tx/0x941970f869c3cda51e5451e9730c6cd6a8f01f9e78894474a26a24fef5b2fd56
After used HOP SwapAndSend USDT 4 times (20, 28, 28, 29 USDT) from Nov-23-2021 to Nov-27-2021
https://polygonscan.com/txs?a=0x0b8181cc4f17d4243d9c88779bb713265791e903&f=2&ps=100&p=1

0xea9ebe9aeff712e20edcbd47b238655631bd226e first used HOP swapAndSend 1096 USDT Sep-20-2021 https://polygonscan.com/tx/0x2c8903620b221bc20171e400f656d3883b3e1e3650198c56de7c070b2c90f05d
After used HOP SwapAndSend USDT 4 times (29, 24, 27, 25 USDT) from Nov-23-2021 to Nov-27-2021
https://polygonscan.com/txs?a=0xea9ebe9aeff712e20edcbd47b238655631bd226e&f=2&ps=100&p=1

Examples for last 5 wallets:
0xbfa31ebe9b1112c64278112090ba2e2130418704 first used HOP swapAndSend 183 USDT Sep-23-2021 https://polygonscan.com/tx/0x94df4c608955cfddf947fd8b9724780a206c5847c70657cbc19416735500f09c
After used HOP SwapAndSend USDT 4 times (22, 22, 25, 30 USDT) from Nov-24-2021 to Nov-27-2021
https://polygonscan.com/txs?a=0xbfa31ebe9b1112c64278112090ba2e2130418704&f=2&ps=100&p=1

0x40b64b354f2079c8aac9320f8f2b386fffca423c first used HOP swapAndSend 991 USDT Sep-22-2021 https://polygonscan.com/tx/0xa15686e75f4765359e328dfedecb435179aee89c4255d76e5417165624c94526
After used HOP SwapAndSend USDT 4 times (23, 22, 22, 24 USDT) from Nov-24-2021 to Nov-27-2021
https://polygonscan.com/txs?a=0x40b64b354f2079c8aac9320f8f2b386fffca423c&f=2&ps=100&p=1

0x8e5ed166aa89a3c98719aeb7348ddcceef7e39bb first used HOP swapAndSend 920 USDT Sep-22-2021 https://polygonscan.com/tx/0x9e5e425869ea17cfa17f7ca6bd973d43b3d4cba6563c73d3e6f065bad9fa7299
After used HOP SwapAndSend USDT 4 times (26, 25, 24, 24 USDT) from Nov-24-2021 to Nov-27-2021
https://polygonscan.com/txs?a=0x8e5ed166aa89a3c98719aeb7348ddcceef7e39bb&f=2&ps=100&p=1

0x2316a6b76dbf0ba68073d972d5a42a050501c585 first used HOP swapAndSend 1103 USDT Sep-22-2021 https://polygonscan.com/tx/0x628241c31b4e9f874d68987112f30240d3da2db2d8c06a0ea73a0408b9ce5060
After used HOP SwapAndSend USDT 4 times (23, 24, 20, 24 USDT) from Nov-24-2021 to Nov-27-2021
https://polygonscan.com/txs?a=0x2316a6b76dbf0ba68073d972d5a42a050501c585&f=2&ps=100&p=1

0x5684f23a3a8f9a367612145ac05d0dad96ccf5da first used HOP swapAndSend 1273 USDT Sep-22-2021 https://polygonscan.com/tx/0x0ab88fad052f76feb0af9a7187b93624a0db4fbde2b4734fc506d2c8fd7dbf22
After used HOP SwapAndSend USDT 4 times (23, 26, 21, 29 USDT) from Nov-24-2021 to Nov-27-2021
https://polygonscan.com/txs?a=0x5684f23a3a8f9a367612145ac05d0dad96ccf5da&f=2&ps=100&p=1

Links for remaining addresses:
https://polygonscan.com/txs?a=0xad1cb13cb3db4694c41a1a91bf4e2e37ef4c2ad1&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x4939ca5dd5364bad45321b90aece36855d823ce5&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xe5865632439987c02d6634ed8335948859fead59&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x58371039b7314b7dd9aae1c43c9fc2b001deb996&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x20b8152ff80c8e7e8a56d47f428a08c054a8ee56&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xab2227c706ccf94bfcfa9b55f0933b3803c0cda9&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x6509f59753b1239322760a6a995b5465a7ca501a&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x89f360b9deae4e0f3ec2bfda31e212e2d0567776&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xa018924e0d156a13f888494659450e6fcaa9ac28&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xdf727d07ebd49237d3f1f34f79b63860139a6c7c&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xbb983df78c0a7fde3c58dc0caca3a2cd5c7868e8&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xe01ba9d1143306e29fbd39fbd5e1cf8cb44a864d&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xdb77dbc91b48f758a332b8961407ca1d22c8292c&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x1bd89d24d3435ad9290e528e4f56472d3468de00&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x7fefd2303b2260abcd97609d48cdd5b61fbb5fdc&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xf8389b647ec7331c64b3c3c91fce93cffb7b075b&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xe15967327932a7ae8d6334f079bd889e20c608b2&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xb5c9b2f2043230b4b86b2884724263b93a299009&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xbb0d240bc794514fc84287ca4ad77282810734cb&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x54cb7407b7a6eca9715bfb3a467ff056acef2d01&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xecf08e10706832af66adec7702aa9eff0fdc878d&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xe2b09bf03404e09007325ebf49b8633848121b5d&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x0068c58783a29256260bb807dfad4807b7193087&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x0331de7345722f62fb31dee19cf6891e13544c63&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xe9c3a8a0ee3e5faf36bbf02d228cf6a2a767213e&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xa57ea970d1f25975640f0d81bf613287b6aa62ea&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xe2118286257a56b864039bbfb90bf3658d13c3d1&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x62d644a6e42e7f93c9704cf456f1fc3d15757ce4&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xea00e2ab1c6878e04ba7f0e310eeb78836bee076&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xe55082fd99bbbd79ec991309417c756e9c1fede7&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xed10ce0f55287e96fa8ef6809b2c948981117f52&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x3fa1b088af50d6b04cf043d46d2bc1e3f239700b&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x7562207875419a6951da5663681ecf611d94f576&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x0495a14956a926f1d8f8176822455bf3476178cd&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xa578487a88f3ef1c3d9788cf0ceb3fb5460f2ba2&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x994831005283888124443a687cf3731ab59428e7&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xc71cdefcdef9572417a4a923e4ca80d3ff237327&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x97c961dfd3fffc4e2980f7ec6402e49640683d4f&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x8c849d6cedf61bd72a0672a54f1a99d1c339e5ab&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x7f97ba1c342f713420a4625772b88d4d38956bd2&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x14ccf97bcd25e5c875b367e93cbfde93894db98f&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0x7c7279711d010e84089df83dc8a3ec91656f4acd&f=2&ps=100&p=1
https://polygonscan.com/txs?a=0xe5af587795c94162028352a816db691ebc7ad71f&f=2&ps=100&p=1

# Methodology

I checked polygon's matic transactions with polygonscan API and filtered transactions, where 20+ recipients addresses and sende are eligible for the airdrop. Then I checked when and how this wallets interact with HOP. Node js code available on request.

# Rewards Address

0x7d3f6048091aEdE6198736b8C6ac527c314e25ff

**Comment by rotate-eth** [comment]
Hi, sorry to say I just reported this one 10 mins before you. https://github.com/hop-protocol/hop-airdrop/issues/189
Looks like our address list is slightly different than mine though. This cluster is tricky to map.

**Comment by GreyPK** [comment]
Yea, identical addresses

**Comment by GreyPK** [comment]
Closed because the addresses are the same as #189
